dap_aria_mapping/utils/novelty_utils.py

```diff
@@ -281,3 +281,72 @@ def aggregate_document_commonness(
     (default: 10th percentile as in the original paper)
     """
     return np.percentile(commonness_scores, percentile)
+
+
+def document_to_topic_novelty(
+    document_novelty: pd.DataFrame,
+    id_column: str = "work_id",
+    aggregation: str = "median",
+) -> pd.DataFrame:
+    """
+    Explodes the topics column and aggregates the novelty scores
+
+    Args:
+        document_novelty (pd.DataFrame): A dataframe with novelty scores and list of topics for each document
+        aggregation (str, optional): The aggregation to use. Defaults to "median".
+
+    Returns:
+        pd.DataFrame: A dataframe with "topic", "year", "doc_counts" and "topic_doc_novelty" novelty score
+    """
+    return (
+        document_novelty.explode("topics")
+        .groupby(["topics", "year"], as_index=False)
+        .agg(
+            topic_doc_novelty=("novelty", aggregation), doc_counts=(id_column, "count")
+        )
+        .rename(columns={"topics": "topic"})
+    )
+
+
+def pair_to_topic_novelty(
+    topic_pair_commonness: pd.DataFrame,
+    aggregation: str = lambda x: aggregate_document_commonness(x),
```

Comment:
woah, a lambda function as default value. In fact, doesn't this return the lambda function itself, not the outputs? I think getting the latter would require a check whether the `aggregation` argument is callable, and then run `aggregation()` within the function.

Reply:
Yeah, I mean syntactically it makes sense, and the behavior is as expected when calling it in `agg`. It's just that not only is a bit unorthodox, but it makes it confusing when the typing hint is `str` and the description in the docstring is vague. Maybe making that a bit neater will work just fine?

Reply:
Thanks a lot, I will change the default to a string `10th_percentile` and then add and if-then statement that defines the lambda function later on. Will be more transparent!